### PR TITLE
Http/2 parser chunks tests

### PR DESCRIPTION
**Pull request by alegz-filomaphitskay**
Contributes to https://github.com/tempesta-tech/tempesta/issues/1297

### CODE REVIEW

File: fw/t/unit/test_http_parser.c [hunk 1]
```diff
@@ -330,75 +326,73 @@ validate_data_fully_parsed(int type)
...
-	    || !validate_data_fully_parsed(type))		\
-		TEST_FAIL("can't parse %s (code=%d):\n%s",	\
-			  (type == FUZZ_REQ			\
-			   || type == FUZZ_REQ_H2		\
-			   ? "request" : "response"),		\
-			  _err, (str)); 			\
-	__fpu_schedule();					\
-	type == FUZZ_REQ_H2 ? false:				\
-		_err == TFW_PASS;				\
+#define TRY_PARSE_EXPECT_PASS(str, type, chunk_mode)					\
```
Comment (ttaym) [inline fw/t/unit/test_http_parser.c:329]: Please, make sure that on the line and everywhere in PR your line width less than or equals to 81.

File: fw/t/unit/test_http_parser.c [hunk 2]
```diff
@@ -148,14 +153,20 @@ split_and_parse_n(unsigned char *str, int type, size_t len, size_t chunk_size)
...
 
 		pos += chunk_size;
 		hm->msg.len += parsed;
 
+		BUILD_BUG_ON((int)TFW_POSTPONE != (int)T_POSTPONE);
 		if (r != TFW_POSTPONE)
 			return r;
+
+		if (type == FUZZ_REQ_H2)
+			r = tfw_h2_parse_req_finish(req);
```
Comment (ttaym) [inline fw/t/unit/test_http_parser.c:169]: There is drastic logic error here. `tfw_h2_parse_req` never return `T_OK`, but `T_DROP` or `T_POSTPONE`. Here you unconditionally call `tfw_h2_parse_req_finish` that almost always returns `T_OK`. In fact all your h2 tests below not even fully ran.

**Review by ttaym — CHANGES_REQUESTED** [review]
Please see my comments about h2 test never fully run and try to fix issue.

And please make sure with logging, debugging or somewhat, that tests in fact runs, especially chunked. Give here tangible evedence of it. See concerns for alexander in the issue description.

File: fw/t/unit/test_http_parser.c (hunk 2 above)
Comment (alegz-filomaphitskay) [reply fw/t/unit/test_http_parser.c:169]: OK. It seems you are right. I'll try to fix this.

File: fw/t/unit/test_http_parser.c (hunk 1 above)
Comment (alegz-filomaphitskay) [reply fw/t/unit/test_http_parser.c:329]: Ok.

**Review by ttaym — APPROVED** [review]
Now i cannot see any defect. LGTM.

But, please, _before_ you merge the PR ensure all tests actually run. See my previous review comment.

It surprises me alot that chunked h2 parser tests pass. So please see my comment on the issue https://github.com/tempesta-tech/tempesta/issues/1297#issuecomment-1035017787 and ensure that there are no problems in `tfw_hpack_decode()` left undiscovered by tests.

When you do all checks please use `Squash and merge` merge option.

**Comment by alegz-filomaphitskay** [comment]
> It surprises me alot that chunked h2 parser tests pass. So please see my comment on the issue [#1297 (comment)](https://github.com/tempesta-tech/tempesta/issues/1297#issuecomment-1035017787) and ensure that there are no problems in `tfw_hpack_decode()` left undiscovered by tests.
> 
I took a closer look at the logs and came to the conclusion that the code is work.
Unfortunately I couldn't make the logs more informative for this task.

**Pull request merged**